genstudio-mlr-claims-app/src/genstudiopem/web-src/src/utils/claimsValidation.ts

```diff
@@ -21,19 +21,32 @@ import { Experience } from '@adobe/genstudio-uix-sdk';
 import { TEST_CLAIMS } from '../Constants';
 import { Key } from 'react';
 
-
 export const claimStatus = {
     Valid: 'valid',
     Violated: 'violated',
     N_A: 'n/a'
+};
+
+const maxCharacterLimits = {
+    header: 80,
+    pre_header: 100,
+    body: 150,
 }
 
-function checkClaim(text: string, claim: string, threshold: number = 0.2): "valid" | "violated" | "n/a" {
+function checkClaim(fieldName: string, text: string, claim: string): typeof claimStatus[keyof typeof claimStatus] {
+    // Check if field has a character limit and if text exceeds it
+    const extractedFieldName = fieldName.replace(/pod\d+_/, '')
+    const hasLimit = extractedFieldName in maxCharacterLimits;
+    const limit = hasLimit ? maxCharacterLimits[extractedFieldName as keyof typeof maxCharacterLimits] : 0;
+    if (hasLimit && text.length > limit) {
+        return claimStatus.Violated;
+    }
```

Comment:
the issue is this isn't a "claim" violation. it is just a "violation". so you need to make sure the message (claimStatus) we show in this case is not "claim X was violated" but instead "violated, max length of X should be Y"

Reply:
updated